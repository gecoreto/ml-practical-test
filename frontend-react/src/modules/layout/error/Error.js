import React from 'react'
import './error.sass'

export const Error = () => {
  return (
    <div className="app-error-container container p-4">
      <h1>Estamos presentando incovenientes</h1>
      <p className="zoom-area">
        <b>Oh no!</b>
        Ha ocurrido un error. (Mapeo de errores...)
    </p>
      <section className="error-container">
        <span className="four">
          <span className="screen-reader-text">4</span>
        </span>
        <span className="zero">
          <span className="screen-reader-text">0</span>
        </span>
        <span className="four">
          <span className="screen-reader-text">4</span>
        </span>
      </section>
      <div className="link-container">
        <button
          className="ml-btn-primary">
          Volver al inicio
        </button>
      </div>
    </div >

  )
}

export default Error