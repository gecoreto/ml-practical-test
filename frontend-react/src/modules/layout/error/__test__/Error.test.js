import React from 'react';
import { shallow } from 'enzyme';
import { Error } from '../Error';

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => {
    return {
        useHistory: () => ({
            push: mockHistoryPush
        })
    };
});
jest.mock("react-redux", () => {
    return {
        useSelector: () => jest.fn(),
        useDispatch: () => jest.fn()
    };
});

describe("Error component", () => {

    const wrapper = shallow(<Error />);

    test('should test Error component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe("WHEN the Button is pressed", () => {

        it("THEN getProductList should be called", () => {
            const event = {
                preventDefault() { },
                target: { value: 'the-value' }
            };
            wrapper.find(`[id="bnt-to-home"]`).simulate('click');
            expect(mockHistoryPush).toHaveBeenCalledWith('/');
        });
    });

});
