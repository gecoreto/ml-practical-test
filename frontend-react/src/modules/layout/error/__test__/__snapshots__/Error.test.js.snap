// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component should test Error component 1`] = `ShallowWrapper {}`;
